cannon/mipsevm/mips.go

```diff
@@ -179,6 +179,47 @@ func (m *InstrumentedState) handleSyscall() error {
 	return nil
 }
 
+func (m *InstrumentedState) pushStack(target uint32) {
+	if !m.debugEnabled {
+		return
+	}
+	m.debug.stack = append(m.debug.stack, target)
+	m.debug.caller = append(m.debug.caller, m.state.PC)
+}
+
+func (m *InstrumentedState) popStack() {
+	if !m.debugEnabled {
+		return
+	}
+	if len(m.debug.stack) != 0 {
```

Comment:
Do we want to log an error message if we try to pop an empty stack?

Reply:
Yes. That's a good idea. Lemme try that; to make sure that this doesn't generate too many spurious errors.